### PR TITLE
Feature/real ocr

### PR DESCRIPTION
**Pull request by TinyMurky**
### DEVELOP
/company/:companyId/OCR/upload
/company/:companyId/OCR/:resultId/process_status
/company/:companyId/OCR/:resultId/result
/company/:companyId/voucher
/company/:companyId/voucher/:voucherId/process_status
/company/:companyId/voucher/:voucherId
以上api串接道AICH

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### CODE REVIEW

**Review by Luphia — CHANGES_REQUESTED** [review]

File: src/constants/config.ts [hunk 1]
```diff
@@ -133,3 +133,9 @@ export const ALLOWED_ORIGINS = [
 ];
 
 export const FIDO2_USER_HANDLE = 'iSunFA-User';
+
+export const FORMIDABLE_CONFIG = {
+  uploadDir: 'tmp',
+};
+
+export const AICH_URI = process.env.AICH_URI || 'http://localhost:3001';
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/constants/config.ts:141]: 修改如下
```ts
export const AICH_URI = process.env.AICH_URI as string;
```

File: src/lib/utils/errorCode.ts
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/lib/utils/errorCode.ts:1]: 更改完成

File: src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts [hunk 2]
```diff
@@ -20,23 +21,20 @@ export default function handler(req: NextApiRequest, res: NextApiResponse<Respon
...
-          fee: 0,
-        },
-      };
+      const result = await fetch(`${AICH_URI}/api/v1/ocr/${resultId}/result`);
+
+      if (!result.ok) {
+        return res.status(500).json({
+          powerby: `ISunFa api ${version}`,
+          success: false,
+          code: '500',
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts]: 目前改成throw error的方法，從error code mapping constant

File: src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts [hunk 3]
```diff
@@ -20,23 +21,20 @@ export default function handler(req: NextApiRequest, res: NextApiResponse<Respon
...
-          hasTax: true,
-          taxPercentage: 25,
-          hasFee: false,
-          fee: 0,
-        },
-      };
+      const result = await fetch(`${AICH_URI}/api/v1/ocr/${resultId}/result`);
+
+      if (!result.ok) {
+        return res.status(500).json({
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts]: 目前改成throw error的方法，從error code mapping constant

File: src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts [hunk 4]
```diff
@@ -20,23 +21,20 @@ export default function handler(req: NextApiRequest, res: NextApiResponse<Respon
...
+          success: false,
+          code: '500',
+          message:
+            'Internal Server Error in ocr get result api, error in fetching OCR result from AICH',
+        });
+      }
+
+      const ocrResultData: AccountInvoiceData = (await result.json()).payload;
+
       return res.status(200).json({
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts]: 目前改成從responseStatusCode.success 取出200的status code

File: src/pages/api/v1/company/[companyId]/voucher/index.ts [hunk 5]
```diff
@@ -6,82 +9,114 @@ import { NextApiRequest, NextApiResponse } from 'next';
...
         message: 'create voucher',
-        payload: voucher,
+        payload: resultStatus,
       });
     } else {
-      throw new Error('Method Not Allowed');
+      res.status(405).json({
+        powerby: 'ISunFa api ' + version,
+        success: false,
+        code: '405',
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/voucher/index.ts]: 目前改成throw error的方法，從error code mapping constant

File: src/pages/api/v1/company/[companyId]/voucher/index.ts [hunk 6]
```diff
@@ -6,82 +9,114 @@ import { NextApiRequest, NextApiResponse } from 'next';
...
         powerby: 'ISunFa api ' + version,
         success: true,
         code: '200',
         message: 'create voucher',
-        payload: voucher,
+        payload: resultStatus,
       });
     } else {
-      throw new Error('Method Not Allowed');
+      res.status(405).json({
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/voucher/index.ts]: 目前改成throw error的方法，從error code mapping constant

File: src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/ocr/[resultId]/result.ts]: 目前改成從responseStatusCode.success 取出200的status code

File: src/pages/api/v1/company/[companyId]/voucher/index.ts [hunk 7]
```diff
@@ -6,82 +9,114 @@ import { NextApiRequest, NextApiResponse } from 'next';
...
       }
-      const voucher: IVoucher = {
-        voucherIndex: '1',
-        metadatas,
-        lineItems,
-      };
+
+      const resultStatus: AccountResultStatus = await result.json();
+
       res.status(200).json({
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/voucher/index.ts]: 目前改成從responseStatusCode.success 取出200的status code

File: src/pages/api/v1/company/[companyId]/voucher/index.ts [hunk 8]
```diff
@@ -6,82 +9,114 @@ import { NextApiRequest, NextApiResponse } from 'next';
...
         powerby: 'ISunFa api ' + version,
         success: true,
         code: '200',
         message: 'create voucher',
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/voucher/index.ts:97]: 目前改成從responseStatusCode.success 取出200的status code

File: src/pages/api/v1/company/[companyId]/voucher/index.ts [hunk 9]
```diff
@@ -6,82 +9,114 @@ import { NextApiRequest, NextApiResponse } from 'next';
...
-        metadatas,
-        lineItems,
-      };
+
+      const resultStatus: AccountResultStatus = await result.json();
+
       res.status(200).json({
         powerby: 'ISunFa api ' + version,
         success: true,
         code: '200',
```
[the remark this answers is not in the document]
Comment (TinyMurky) [reply src/pages/api/v1/company/[companyId]/voucher/index.ts]: 目前改成從responseStatusCode.success 取出200的status code

**Pull request closed without merging**